### DISCUSSION TITLE
Using Joule Heating Source to simulate heat amount (power loss) of a cable

### DISCUSSION THREAD

**Discussion by AgentKTU** [idaholab/moose#29588] [Q&A General]
### Check these boxes if you have followed the posting rules.

- [X] Q&A General is the most appropriate section for my question
- [X] I have consulted the posting Guidelines on the Discussions front page
- [X] I have searched the Discussions forum and my question has not been asked before
- [X] I have searched the MOOSE website and the documentation does not answer my question
- [X] I have formatted my post following the posting guidelines (screenshots as a last resort, triple back quotes around pasted text)

### Question

Hello everyone,

I am trying to use the `JouleHeatingSource` to Calculate the heat source term corresponding to electrostatic Joule heating for an underground power cable. The given example Inputfile on the Moose-website runs with a given voltage value (BC) for the left and right boundary of a simple 2d copperplate geometry.

(https://mooseframework.inl.gov/source/kernels/JouleHeatingSource.html)

In general: 
Is it possible to use a given current value instead of a voltage value for the `elec` variable, to calculate the heat source term? 
What is the output unit of the `Joule Heating Source` or the variable `elec` --> I assumed [W] ???

I tried applying the example Inputfile on to my cable geometry (simple 2D circle, generated with Gmsh) where I defined a physical point as a point-source (e.g. 0.1 V) and an outer boundary circle with 0 V. The material properties are assigned to the created surface, see mesh & Inputfile below. (Pleas let me know if the Link doesn't work.)

[https://github.com/AgentKTU/JH-Power-cable.git](url)

Using this approach I get a calculated temperature of 700 K at the outer boundary, that doesn't seem realistic.

I tried saving the output of the JouleHeatingSource with   `save_in = var`. (see `AuxVariable` in input file) but the results aren't reasonable. The `var` values are all negativ.
What is the assumed unit or value of the `JouleHeatingModule`???

My goal is to export the value of the Power loss (the heat source term) and implement it to my PorousFlow soil model where I added a heat source boundary.
At the end I want to simulate the temperature distribution in the soil by defining realistic boundary conditions and see how the cable temperature reacts.

I am happy about any help!
Konstantin

**Comment by GiudGiud** [comment]
> I tried saving the output of the JouleHeatingSource with save_in = var. (see AuxVariable in input file) but the results aren't reasonable.

Note that this option gave you the contribution to the residual of that kernel. This can be a little obscure. 
They are negative because all terms get moved to the left hand side of the equation, so "sources" are negative, while losses are positive

> Is it possible to use a given current value instead of a voltage value for the elec variable, to calculate the heat source term?

@cticenhour can we do this?

> What is the output unit of the Joule Heating Source or the variable elec --> I assumed [W] ???

if the mesh and all quantities are in SI units, then the units of the variables and any input to the kernel should also be in SI units

**Reply by cticenhour** [reply]
Not as-designed with the JouleHeatingSource object as it is, as we make an assumption that the current $\mathbf{J} = \sigma \cdot \mathbf{E}$, where $\sigma$ is the electrical conductivity and $\mathbf{E}$ is the electrostatic electric field, which can be defined as $\mathbf{E} = - \nabla \phi$ where $\phi$ is the electrostatic potential. So, the heat source $q$ that we're adding to the PDE using JouleHeatingSource is:

$q = \mathbf{J} \cdot \mathbf{E} = \sigma \cdot \mathbf{E} \cdot \mathbf{E} = \sigma \cdot \nabla \phi \cdot \nabla \phi$

with standard SI units of Watts per meters-cubed. The variable `elec` used in the object would normally be in units of Volts. We don't make explicit assumptions on units in many MOOSE objects, so as long as you are consistent in your input file, we don't impose a specific unit system on you. 

To go back to the question of whether we can use a current value or not, I cannot think of an alternative off the top of my head that you could use with the right mathematical form. @GiudGiud do you have a thought on this? Maybe we're lucky, and I missed it in the EM module and the framework when I looked through just now. In any case, your residual comtribution (in `computeQpResidual`) for such an object would need to be something like:

```
-_current[_qp] * (-_grad_elec[_qp]) * _test[_i][_qp];
```

Where you have a new object parameter to represent your coupled current value (if a variable), or the function / material property form that you might use to compute it otherwise.

**Reply by GiudGiud** [reply]
I think this makes sense. If current is the main variable you solve for, then use a variable for it. Else I would recommend using an AD material property.

> In any case, your residual comtribution (in computeQpResidual) for such an object would need to be something like:

This should be all you need to create a new AD kernel tbh so you might as well do that.
